### PR TITLE
ploigos-tool-containers - add s2i

### PR DESCRIPTION
**Pull request by itewk**
# Purpose
add s2i to ploigos-tool-containers

# Breaking?
No

# Integration Testing
* [x] [Minimal](https://jenkins-jenkins.apps.tssc.rht-set.com/job/Ploigos%20Reference%20Applciations%20(minimal)/job/reference-undertow-mvn-s2i/view/change-requests/job/PR-1/20/)

### CODE REVIEW

**Comment by dwinchell** [comment]
@itewk looks good. why did we need this?

**Comment by cthiel1** [comment]
@dwinchell CMS needs it. The dev team we want to onboard is using S2I builds, not container files, and we're going to work to build that into the psr over the next few days

File: .github/workflows/publish.yaml [hunk 1]
```diff
@@ -1869,6 +1869,9 @@ jobs:
           echo "test buildah"
           docker run -u 1001 ${{ env.IMAGE_TAG_LOCAL }} buildah --version
 
+          echo "test s2i"
```
Comment (cthiel1) [inline .github/workflows/publish.yaml:1872]: can we get rid of this?

**Review by cthiel1 — APPROVED** [review]
Nitpicky request to remove (extra?) echo, but otherwise looks good

File: .github/workflows/publish.yaml (hunk 1 above)
Comment (dwinchell) [reply .github/workflows/publish.yaml:1872]: that was my first reaction, but i think we actually want it. If you look above in the file, there are a bunch of echos for similar things. They show up in the output of the publish pipeline as status updates.

**Review by dwinchell — APPROVED** [review]

**Review by dwinchell — APPROVED** [review]

**Comment by itewk** [comment]
had to update the base ploigos-tool-container image to have a podman socket since s2i expects a dockersocket to interact with.

**Comment by itewk** [comment]
> had to update the base ploigos-tool-container image to have a podman socket since s2i expects a dockersocket to interact with.

actually. wrong. so s2i only works with docker when doing the container build. https://cloud.redhat.com/blog/openshift-4-image-builds

BUUUUT

there is `s2i build . registry.redhat.io/ubi8/php-74:latest --dockercfg-path ~/container-registry-auth --as-dockerfile Containerfile` which will build a container file that you can then do `buildah bud Containerfile` sooo....we have a pretty clean path forward. have an s2i step implementer that crteates a containerfile, then call the existing buildah stepimplmeneter to build that contaienrfile. boom, bang done.

**Pull request merged**